airflow/settings.py

```diff
@@ -278,17 +280,29 @@ def remove(*args, **kwargs):
         pass
 
 
+def _is_sqlite_db_path_relative(sqla_conn_str: str) -> bool:
+    """Determine whether the database connection URI specifies an absolute path."""
+    import re2
+
+    is_abs_win_prefix: bool = sys.platform == "win32" and bool(
+        re2.match(r"^sqlite:///" + _WINDOWS_ABSOLUTE_PATH_PREFIX, sqla_conn_str)
+    )
+    is_abs_unix_prefix: bool = sqla_conn_str.startswith("sqlite:////")
```

Comment:
You can instead use `os.path.isabs` after removing the prefix. The platform check is also not necessary that way.

Reply:
@uranusjr I applied your suggestion to `settings.py`, but couldn't use it to similarly simplify the tests.  It seems like the output of `isabs` for Windows paths depends on the host OS:

Python 3.12 @ Windows:
![image](https://github.com/apache/airflow/assets/6509619/fbd59682-3bf1-4b5e-aa7f-22d2273a9eb6)

---
Python 3.9 & 3.11 @ Linux
![image](https://github.com/apache/airflow/assets/6509619/2fdc8268-160f-4226-b0e9-4f3a90a48a86)